cli/src/main/java/com/devonfw/tools/ide/merge/xmlmerger/XmlMerger.java

```diff
@@ -151,41 +131,68 @@ public void inverseMerge(Path workspace, EnvironmentVariables variables, boolean
     if (!Files.exists(workspace) || !Files.exists(update)) {
       return;
     }
-    Document updateDocument = load(update);
-    Document workspaceDocument = load(workspace);
-    resolve(updateDocument, variables, true, workspace.getFileName());
-    MergeStrategy strategy = MergeStrategy.OVERRIDE;
-    MergeElement sourceRoot = new MergeElement(workspaceDocument.getDocumentElement(), workspace);
-    MergeElement targetRoot = new MergeElement(updateDocument.getDocumentElement(), update);
-    strategy.merge(sourceRoot, targetRoot, null);
-    save(updateDocument, update);
-    this.context.debug("Saved changes in {} to {}", workspace.getFileName(), update);
+    throw new UnsupportedOperationException("not implemented!");
+  }
+
+  /**
+   * {@link #load(Path) Loads} and {@link #resolveDocument(XmlMergeDocument, EnvironmentVariables, boolean) resolves} XML from the given file.
+   *
+   * @param file the {@link Path} to the XML file.
+   * @param variables the {@link EnvironmentVariables}.
+   * @return the loaded {@link XmlMergeDocument}.
+   */
+  public XmlMergeDocument loadAndResolve(Path file, EnvironmentVariables variables) {
+
+    XmlMergeDocument document = load(file);
+    resolveDocument(document, variables, false);
+    return document;
   }
 
-  public Document load(Path file) {
+  /**
+   * @param file the {@link Path} to the XML file.
+   * @return the loaded {@link XmlMergeDocument}.
+   */
+  public XmlMergeDocument load(Path file) {
 
     try (InputStream in = Files.newInputStream(file)) {
-      return DOCUMENT_BUILDER.parse(in);
+      Document document = DOCUMENT_BUILDER.parse(in);
+      return new XmlMergeDocument(document, file);
     } catch (Exception e) {
       throw new IllegalStateException("Failed to load XML from: " + file, e);
     }
   }
 
+  /**
+   * @param document the XML {@link XmlMergeDocument} to save.
+   */
+  public void save(XmlMergeDocument document) {
+
+    save(document.getDocument(), document.getPath());
+  }
+
+  /**
+   * @param document the XML {@link Document} to save.
+   * @param file the {@link Path} to the file where to save the XML.
+   */
   public void save(Document document, Path file) {
 
     ensureParentDirectoryExists(file);
     try {
+      //Source xsltSource = new StringSource(xslt);
+      //Templates templates = TRANSFORMER_FACTORY.newTemplates(xsltSource);
+      //Transformer transformer = templates.newTransformer();
```

Comment:
```suggestion
```

Reply:
Adjusted.

Reply:
I would like to remove the `merge` namespace and the indent whitespaces via a transformer stylesheet.
IMHO this would be more clean and efficient.
I will create a new issue for it.